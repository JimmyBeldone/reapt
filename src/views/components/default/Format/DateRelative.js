/* eslint-disable consistent-return */
import React from "react";
import PropTypes from "prop-types";
import dayjs from "dayjs";
import relativeTime from "dayjs/plugin/relativeTime";

dayjs.extend(relativeTime);

const DateRelative = ({ className, value, relativeTo, relativeNow }) => (
    <div className={className}>
        {relativeNow
            ? relativeTo
                ? dayjs().fromNow()
                : dayjs().toNow()
            : relativeTo && value !== null
            ? dayjs().from(dayjs(value))
            : dayjs().to(dayjs(value))}
    </div>
);

DateRelative.propTypes = {
    className: PropTypes.string,
    value: PropTypes.string.isRequired,
    relativeTo: PropTypes.bool,
    relativeNow: PropTypes.bool
};

DateRelative.defaultProps = {
    className: "",
    relativeTo: true,
    relativeNow: false
};

export default DateRelative;
