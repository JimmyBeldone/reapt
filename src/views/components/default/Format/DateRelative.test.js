import React from "react";
import { render } from "enzyme";
import dayjs from "dayjs";
import DateRelative from "./DateRelative";
import moment from "moment-timezone";

describe("Test DateRelative Component", () => {
    it("Should render date relative from another date", () => {
        const wrapper = render(<DateRelative value="2015-24-01" />);
        expect(wrapper.text()).toBe("in 2 years");
    });

    it("Should render date relative to another date", () => {
        const wrapper = render(
            <DateRelative relativeTo={false} value="2015-24-01" />
        );
        expect(wrapper.text()).toBe("2 years ago");
    });

    it("Should render date relative from now", () => {
        jest.mock("moment", () => () => ({
            format: () => "2018–01–30T12:34:56+00:00"
        }));
        const wrapper = render(<DateRelative relativeNow />);
        expect(wrapper.text()).toMatchSnapshot();
    });

    it("Should render date relative to now", () => {
        const wrapper = render(<DateRelative relativeNow relativeTo={false} />);
        expect(wrapper.text()).toBe("a few seconds ago");
    });
});
