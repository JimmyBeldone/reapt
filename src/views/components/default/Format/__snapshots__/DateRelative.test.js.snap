// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DateRelative Component Should render date relative from now 1`] = `"a few seconds ago"`;
